src/MICore/Transports/LocalUnixTerminalTransport.cs

```diff
@@ -2,16 +2,26 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Globalization;
 using System.IO;
 using System.Text;
+using System.Threading.Tasks;
 
 namespace MICore
 {
     public class LocalUnixTerminalTransport : StreamTransport
     {
         private string _dbgStdInName;
         private string _dbgStdOutName;
-        private FileSystemWatcher _fifoWatcher;
+        private int _debuggerPid = -1;
+
+        /// <summary>
+        /// When the debugger exits, it will write into the fifo associated with this reader.
+        /// This is useful in the case of attach when the root password is required, but the
+        /// request is canceled. In that case, the debugger won't have started yet (so no messages
+        /// will be processed from it), but the trap will execute and write into this fifo.
```

Comment:
The original bug the filewatcher trying to address is "killall -9 gdb", make sure VSCode exit debug mode when gdb is killed that way